miraclelinux/meta-debian
# libsdl2: Update version from 2.0.9+dfsg1-1 to 2.0.9+dfsg1-1+deb10u1

# Purpose of pull request

This version fixes following CVEs.
- CVE-2019-7572
- CVE-2019-7573
- CVE-2019-7574
- CVE-2019-7575
- CVE-2019-7576
- CVE-2019-7577
- CVE-2019-7578
- CVE-2019-7635
- CVE-2019-7636
- CVE-2019-7638
- CVE-2019-13616
- CVE-2019-13626
- CVE-2020-14409
- CVE-2020-14410
- CVE-2021-33657
- CVE-2022-4743

DLA: DLA 3314-1

# Test
## How to test
1. Build package
2. Build core-image-weston
Add the following to local.conf and build core-image-weston image.
```
MACHINE = "qemuarm64"
DISTRO = "emlinux"
IMAGE_INSTALL_append = " libsdl2 libsdl2-dev"
EXTRA_IMAGE_FEATURES_append = " tools-sdk"
```
3. Prepare test code to check work of the libsdl2 library.
To create the test code, I used the code from the following URL as a reference.
- https://qiita.com/nanbuwks/items/379013b65fa10ce042a5
- http://sdl2referencejp.osdn.jp/SDL_RenderDrawLine.html

test.c: 
```
#include <SDL2/SDL.h>

int main(int argc, char** argv) {
    SDL_Init(SDL_INIT_VIDEO);
    // 512 x 512 size window
    SDL_Window* window = SDL_CreateWindow("Check SDL2 library works",
      SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, 512, 512,
      SDL_WINDOW_SHOWN | SDL_WINDOW_OPENGL);
    SDL_Renderer* renderer = SDL_CreateRenderer(
      window, -1, SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC);

    SDL_SetRenderDrawColor(renderer, 0, 128, 0, 255); // R:0, G:128, B:0, A:255
    SDL_RenderClear(renderer);

    SDL_SetRenderDrawColor(renderer, 0, 0, 0, 255); // R:0, G:0, B:0, A:255
    // Draw lines to form a triangle
    SDL_RenderDrawLine(renderer, 320, 200, 300, 240);
    SDL_RenderDrawLine(renderer, 300, 240, 340, 240);
    SDL_RenderDrawLine(renderer, 340, 240, 320, 200);
    SDL_RenderPresent(renderer);
    SDL_Delay(3000); // 3 seconds

    SDL_DestroyRenderer(renderer);
    SDL_DestroyWindow(window);
    SDL_Quit();
    return 0;
}
```
4. Compile and run the test code.
```
gcc test.c -o test -lSDL2
./test
```
## Test result
Build succeeded.
```
$ bitbake libsdl2
Loading cache: 100% |########################################################################################################################################| Time: 0:00:03
Loaded 2368 entries from dependency cache.
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |######################################################################################################################################| Time: 0:00:49
Parsing of 1351 .bb files complete (838 cached, 513 parsed). 2368 targets, 80 skipped, 6 masked, 0 errors.
Removing 1 recipes from the aarch64 sysroot: 100% |##########################################################################################################| Time: 0:00:00
Removing 1 recipes from the qemuarm64 sysroot: 100% |########################################################################################################| Time: 0:00:00
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.6"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "update-libsdl2_2.0.9+dfsg1-1+deb10u1:04c61154ed329ce9dddef13db7e0b77207a3e999"
meta-debian-extended = "HEAD:c248eaa440c32789b6a01c3c6f4789c922212c51"
meta-emlinux         = "HEAD:ead20ef617c1af2d70f1e26e27a99ebd10160c80"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

Initialising tasks: 100% |###################################################################################################################################| Time: 0:00:06
Sstate summary: Wanted 92 Found 32 Missed 60 Current 539 (34% match, 90% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2721 tasks of which 2574 didn't need to be rerun and all succeeded.
```
The test code is working as intended.
![Screenshot from 2023-03-03 09-46-50](https://user-images.githubusercontent.com/118246107/222604514-21d76931-12ea-4a3d-954d-84c51a004872.png)
